### PR TITLE
Enable more tests on Python 3 for testing trial

### PR DESCRIPTION
**Pull request by rodrigc**
http://twistedmatrix.com/trac/ticket/8976

### CODE REVIEW

File: src/twisted/trial/_dist/test/test_worker.py [hunk 1]
```diff
@@ -22,7 +22,8 @@
 from twisted.internet.defer import fail, succeed
 from twisted.internet.main import CONNECTION_DONE
 from twisted.internet.error import ConnectionDone
-from twisted.python.compat import _PY3, NativeStringIO as StringIO, unicode
+from twisted.python.compat import (NativeStringIO as StringIO, unicode)
```
Comment (exarkun) [inline src/twisted/trial/_dist/test/test_worker.py]: `()`s on a single-line import look odd.  Not sure if this is following some more general pattern.  I'd probably leave them off, if I were making the change.

File: src/twisted/trial/_dist/test/test_worker.py [hunk 2]
```diff
@@ -22,7 +22,8 @@
...
+from twisted.python.reflect import qual
```
Comment (exarkun) [inline src/twisted/trial/_dist/test/test_worker.py]: `qual` is not deprecated but new code should probably prefer `fullyQualifiedName` instead.

Comment (rodrigc) [reply src/twisted/trial/_dist/test/test_worker.py]: OK

File: src/twisted/trial/_dist/test/test_worker.py (hunk 1 above)
Comment (rodrigc) [reply src/twisted/trial/_dist/test/test_worker.py]: OK

**Comment by rodrigc** [comment]
@exarkun is this OK now?

**Comment by exarkun** [comment]
@rodrigc Ah sorry, I didn't really review this.  I started to and then realized that I _meant_ to be looking at #649.  Sorry. :)

**Comment by rodrigc** [comment]
@exarkun I incorporated your review feedback from https://github.com/twisted/twisted/pull/659#pullrequestreview-16705317 so I think this is good to go.

**Comment by exarkun** [comment]
Hi Craig,

I didn't review or approve this.

Jean-Paul

**Comment by rodrigc** [comment]
I thought this was your review: https://github.com/twisted/twisted/pull/659#pullrequestreview-16705317 ?

**Comment by exarkun** [comment]
Did you see my follow-up comment?

> @rodrigc Ah sorry, I **didn't really review this**. I started to and then realized that I meant to be looking at #649. Sorry. :)

**Comment by rodrigc** [comment]
Oh sorry, I was confused by your comment about https://github.com/twisted/twisted/pull/649, because you actually added valid review comments for this review.

**Comment by exarkun** [comment]
Ah.  Sorry about that.  FWIW, in the future, if I do a review, I'll remove the "review" keyword from the Twisted ticket and if I give approval I'll use the GitHub "Approve" review type.

File: src/twisted/trial/_dist/test/test_worker.py [hunk 3]
```diff
@@ -171,14 +168,10 @@ def test_runErrorWithFrames(self):
...
-        if _PY3:
-            errorClass = b'builtins.ValueError'
-        else:
-            errorClass = b'exceptions.ValueError'
-
+        errorClass = fullyQualifiedName(ValueError)
         d = self.worker.callRemote(managercommands.AddError,
                                    testName=self.testName, error=b'error',
-                                   errorClass=errorClass,
+                                   errorClass=errorClass.encode("ascii"),
```
Comment (exarkun) [inline src/twisted/trial/_dist/test/test_worker.py:174]: `nativeString` might have been a better solution here.

**Pull request merged**